src/GLONASS_IFB_CORRECTION/glonass_IFB_correction.c

```diff
@@ -0,0 +1,412 @@
+/*
+ *   GLONASS INTER-FREQUENCY PHASE BIASES (IFB) CORRECTION MODULE
+ */
+
+#include "glonass_IFB_correction.h"
+
+#define MIN(x,y)    ( (x) <= (y) ? (x) : (y) )
+
+/* -------------------------------------------------------------------------------------------------------------- */
+/* API */
+
+extern void *glo_IFB_init()
+{
+  glo_IFB_t *glo_IFB = malloc(sizeof(glo_IFB_t));
+
+  if ( glo_IFB == NULL ) {
+
+    return NULL;
+  }
+
+  glo_IFB->mode             = GLO_IFB_SEARCH_MODE;
+  glo_IFB->adjustment_count = 0;
+  glo_IFB->fix_outage       = 0;
+  glo_IFB->glo_dt           = 0.0;
+  glo_IFB->glo_dt_initial   = 0.0;
+  glo_IFB->delta_glo_dt     = 0.0;
+  glo_IFB->signal_to_reset  = 0;
+
+  return (void *) glo_IFB;
+}
+
+extern int glo_IFB_is_valid(const void *glo_IFB)
```

Comment:
`is_valid` just shouts "make me return a boolean". As far as I remember we don't need the C89 so why bother?